src/EditorFeatures/Core.Wpf/Lightup/LightupHelpers.cs

```diff
@@ -328,21 +329,26 @@ public static Action<T> CreateActionAccessor<T>(Type? type, string methodName)
             throw new InvalidOperationException($"Method '{method}' produces a value of type '{method.ReturnType}', which is not assignable to type '{typeof(TResult)}'");
         }
 
-        var parameter = Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
-        var argument = Expression.Parameter(typeof(TArg), parameters[0].Name);
-        var instance =
+        var instanceParameter =
             type.GetTypeInfo().IsAssignableFrom(typeof(T).GetTypeInfo())
-            ? (Expression)parameter
-            : Expression.Convert(parameter, type);
+                ? Expression.Parameter(type, GenerateParameterName(type))
+                : Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
+
+        var argument = Expression.Parameter(typeof(TArg), parameters[0].Name);
         var convertedArgument =
             argType.GetTypeInfo().IsAssignableFrom(typeof(TArg).GetTypeInfo())
             ? (Expression)argument
             : Expression.Convert(argument, argType);
 
         var expression =
             Expression.Lambda<Func<T, TArg, TResult>>(
-                Expression.Convert(Expression.Call(instance, method, convertedArgument), typeof(TResult)),
-                parameter);
+                Expression.Convert(
+                    Expression.Call(
+                        Expression.Convert(instanceParameter, type),
+                        method,
+                        convertedArgument), typeof(TResult)),
+                instanceParameter,
+                argument);
```

Comment:
📝 On review, I would have expected the _only_ change to this method to be the addition of `, argument`

Reply:
It should be now. (Add `instace`)